### PR TITLE
amp-amazon-onetag : Amazon affiliate extension

### PR DESCRIPTION
**Pull request by ankikala**
✨ amp-amazon-onetag : Amazon affiliate extension
Solves #23166

### CODE REVIEW

File: extensions/amp-amazon-onetag/0.1/test/validator-amp-amazon-onetag.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<!doctype html>
```
Comment (Gregable) [inline extensions/amp-amazon-onetag/0.1/test/validator-amp-amazon-onetag.html]: It's a bit silly, but please add a copyright statement to the test cases. You can see other extension tests for an example.

File: extensions/amp-amazon-onetag/validator-amp-amazon-onetag.protoascii [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  extension_spec: {
+    name: "amp-amazon-onetag"
+    version: "0.1"
+    version: "latest"
+  }
+  attr_lists: "common-extension-attrs"
+}
+tags: {  # <amp-amazon-onetag>
+  html_format: AMP
+  tag_name: "AMP-AMAZON-ONETAG"
```
Comment (Gregable) [inline extensions/amp-amazon-onetag/validator-amp-amazon-onetag.protoascii]: Your test examples show a few more attributes: `marketplace` and `instanceid`.  I'm guessing these are intended to be valid. Can you add them in the config for this tag. For example:

```
attrs: {
  name: "instanceid"
}
attrs: {
  name: "marketplace"
}
```

File: extensions/amp-amazon-onetag/0.1/amp-amazon-onetag.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    onetagEndpoints.set("NL", "z-eu.associates-amazon.com");
+    onetagEndpoints.set("SA", "z-eu.associates-amazon.com");
+    onetagEndpoints.set("TR", "z-eu.associates-amazon.com");
+    onetagEndpoints.set("AE", "z-eu.associates-amazon.com");
+
+    var marketplace = this.element.getAttribute('marketplace');
+    var instanceID = this.element.getAttribute('instanceID');
+    if (onetagEndpoints.has(marketplace))  {
+      var scriptURL = "https://" + onetagEndpoints.get(marketplace) + "/widgets/onejs?MarketPlace=" + marketplace + "&adInstanceId=" + instanceID;
+      loadScript(this, scriptURL);
```
Comment (choumx) [inline extensions/amp-amazon-onetag/0.1/amp-amazon-onetag.js]: As @lannka pointed out, we don't allow 3p JS outside of iframes on AMP pages.

**Comment by mrjoro** [comment]
Is this PR still active?

**Comment by vigneshcamar** [comment]
> Is this PR still active?

the PR can be discarded. But the issue is still to be open as we are working on it.

**Pull request closed without merging**